edx_proctoring/api.py

```diff
@@ -279,6 +279,16 @@ def get_exam_attempt_by_id(attempt_id):
     return _get_exam_attempt(exam_attempt_obj)
 
 
+def get_exam_attempt_by_code(attempt_code):
+    """
+    Signals the beginning of an exam attempt when we only have
+    an attempt code
+    """
+
+    exam_attempt_obj = ProctoredExamStudentAttempt.objects.get_exam_attempt_by_code(attempt_code)
+    return _get_exam_attempt(exam_attempt_obj)
```

Comment:
Not related to this code, but the line in _get_exam_attempt()

we should probably check the exam_attempt_obj for None in the start rather than in the second line:

attempt = serialized_attempt_obj.data if exam_attempt_obj else None

otherwise
attempt = serialized_attempt_obj.data if serialized_attempt_obj else None might be better.